extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java

```diff
@@ -189,6 +188,50 @@
         return result;
     }
 
+    /**
+     * If user defines a custom command via configuration, this is used.
+     * If not, it will use the one from other extensions.
+     *
+     * @param target The deployment target (e.g. kubernetes, openshift, knative)
+     * @param name The name of the resource to accept the configuration
+     * @param config The {@link PlatformConfiguration} instance
+     * @param command Optional command item from other extensions
+     */
+    private static List<DecoratorBuildItem> createCommandDecorator(Optional<Project> project, String target, String name,
+            PlatformConfiguration config, Optional<KubernetesCommandBuildItem> command) {
+        List<DecoratorBuildItem> result = new ArrayList<>();
+        if (config.getCommand().isPresent()) {
+            // If command has been set in configuration, we use it
+            result.add(new DecoratorBuildItem(target,
+                    new ApplyCommandDecorator(name, config.getCommand().get().toArray(new String[0]))));
+        } else if (command.isPresent()) {
+            // If not, we use the command that has been provided in other extensions (if any).
+            result.add(new DecoratorBuildItem(target,
+                    new ApplyCommandDecorator(name, command.get().getCommand())));
+        }
+
+        return result;
+    }
+
+    /**
+     * If user defines arguments via configuration, then these will be merged to the ones from other extensions.
+     * If not, then only the arguments from other extensions will be used if any.
+     *
+     * @param target The deployment target (e.g. kubernetes, openshift, knative)
+     * @param name The name of the resource to accept the configuration
+     * @param config The {@link PlatformConfiguration} instance
+     * @param command Optional command item from other extensions
+     */
+    private static List<DecoratorBuildItem> createArgsDecorator(Optional<Project> project, String target, String name,
+            PlatformConfiguration config, Optional<KubernetesCommandBuildItem> command) {
+        List<String> args = new ArrayList<>();
+        command.ifPresent(cmd -> args.addAll(Arrays.asList(cmd.getArgs())));
```

Comment:
Topic for discussion: Since, we are slightly changing the way we perceive `arguments` do we expect extension to contribute to the them? It kind of feels that now extensions are expected to only toy with `command`. Thoughts?

Reply:
For native, extensions still need to provide both the command and the arguments.
But I like what you say which would imply split KubernetesCommandBuildItem in KubernetesCommandBuildItem (only for the command) and KubernetesArgsBuildItem (only for the args).
Are you ok with and do you want me to do it as part of this PR?

Reply:
Why do we need to change `KubernetesCommandBuildItem`? I didn't get that part

Reply:
We don't need to change anything.
I am just wondering / thinking out loud.

Do we expect extensions/build items to ever contribute to arguments?

Reply:
Yes, for Native we need extensions to contribute to the arguments.

Reply:
I don't see why the BuildItem now uses arrrays for both the command and args... Seems counterintuitive...

Reply:
I agree with you, I would change it to only have a single constructor for the command: `KubernetesCommandBuildItem(String ... cmd)` and add a method to set the arguments (as this is now optional for extensions).
Wdyt?

Reply:
Makes sense to me

Reply:
I've already refactored the `KubernetesCommandBuildItem`.